src/JanusGraph.Net/JanusGraphClientBuilder.cs

```diff
@@ -32,21 +32,23 @@ public class JanusGraphClientBuilder
     {
         private readonly GremlinServer _server;
         private readonly JanusGraphSONReaderBuilder _readerBuilder = JanusGraphSONReaderBuilder.Build();
-        private readonly JanusGraphSONWriterBuilder _writerBuilder = JanusGraphSONWriterBuilder.Build();
+        private readonly JanusGraphSONWriterBuilder _writerBuilder;
         private ConnectionPoolSettings _connectionPoolSettings;
 
-        private JanusGraphClientBuilder(GremlinServer server)
+        private JanusGraphClientBuilder(GremlinServer server, bool janusGraphPredicates)
         {
             _server = server;
+            _writerBuilder = JanusGraphSONWriterBuilder.Build(janusGraphPredicates);
         }
 
         /// <summary>
         ///     Initializes a <see cref="JanusGraphClientBuilder" /> for the given <see cref="GremlinServer" />.
         /// </summary>
         /// <param name="server">The <see cref="GremlinServer" /> requests should be sent to.</param>
-        public static JanusGraphClientBuilder BuildClientForServer(GremlinServer server)
+        /// <param name="janusGraphPredicates">This value should be true for JanusGraph-server of version 0.6 and above.</param>
```

Comment:
I think it should be version `0.6.0` as we don't have versions like `0.6` to avoid any confusion and I would add that it should be set to `false` for versions < `0.6.0`. Also it should be _JanusGraph Server_ instead of _JanusGraph-server_.
So, maybe:

> This value activates support for JanusGraph predicate serialization added in JanusGraph 0.6.0. It should be set to `true` for JanusGraph Server versions >= `0.6.0` and to `false` for versions < `0.6.0`.